Package.swift

```diff
@@ -61,6 +66,30 @@ var swiftSettings: [SwiftSetting] { [
     .enableExperimentalFeature("StrictConcurrency=complete"),
 ] }
 
+#if canImport(WASILibc)
+#if canImport(wasi_pthread)
+// Some wasm SDKs provide pthread
+// API's, and some future SDKs will as well.
+//
+// If pthread API's are available, we should use them.
+//
+// But if not, then we can't use them.
+var threadSafetyConfiguration: CSetting { .define("SQLITE_THREADSAFE", to: "1") }
+#else
+// In this case, we're using a Swift for WebAssembly
+// SDK that does NOT have pthread support, so we
+// can't compile in thread safety.
+//
+// Note that by the same principal, the executable will
+// almost certainly be single-threaded, so it is acceptable
+// and fine to elide thread safety. But in this case, there
+// isn't any other option.
+var threadSafetyConfiguration: CSetting { .define("SQLITE_THREADSAFE", to: "0") }
+#endif
+#else
+var threadSafetyConfiguration: CSetting { .define("SQLITE_THREADSAFE", to: "1") }
+#endif
+
```

Comment:
`canImport()` is neither reliable nor safe in `Package.swift` manifests; that's one of the reasons platform conditions exist in the API. I am skeptical that this will work as intended; it might be better to just disable thread-safety unconditionally in Wasm for now.

```suggestion
/// Some wasm SDKs provide pthread API's, and some future SDKs will as well.
/// If pthread API's are available, we should use them.
#if !canImport(WASILibc) || canImport(wasi_pthread)
var threadSafetySetting = "1"
#else
var threadSafetySetting = "0"
#endif

```

Reply:
Done

Reply:
Verified this works by running CI. https://github.com/PassiveLogic/sqlite-nio/actions/runs/20175897833/job/57923753469?pr=1

A misconfiguration causes existing unit tests to fail if thread safety becomes disabled for currently tested platforms, based on my previous testing on this.

The fact that all those tests pass indicates the latest changes properly sets up the configuration.

I also confirmed I can compile and link a wasm binary locally when consuming this latest configuration change.